common-practices-tools/contribution/contrib-first.md

```diff
@@ -0,0 +1,29 @@
+# Contrib First
+
+Whenever we are building something that could be of use by more than one project or client, we build it as contributed work first, if our contract and security concerns allow for it. Contrib First is an approach we take with both patches to FOSS software and new contributions.
+
+## Rationale for contrib first
+
+-   **Fiscal responsibility** - Building it and contributing it means that other government agencies will never have to pay to build the same thing twice. This helps agencies comply with Federal Source Code Policy: Achieving Efficiency, Transparency, and Innovation through Reusable and Open Source Software [OMB Memorandum M-16-21](https://obamawhitehouse.archives.gov/sites/default/files/omb/memoranda/2016/m_16_21.pdf)
+-   **Reusability** - CivcActions other clients can benefit from work that was already done.
+-   **Security** - contributing our work to an open source project like Drupal means it may receive security coverage by the Drupal security team and the public.
+-   **Avoiding the gift that never happens** - Clients are not typically supportive of taking working local software that was already built for them and in use by them, and then paying to move or refactor that software to become open source. The benefit is too small for the cost. By building it as contributed code first, there is no extra cost.
+-   **Development happens in the open** - The issues are public. The commits are public. Everyone can contribute improvements.
+-   **Reliability** - A solution built for contribution is often better designed, and better documented than a local solution meant to "just get it done". By putting our company and personal names on it publicly we commit to a quality product. Releasing a FOSS solution also increases the number of testers and edge cases that can surface and reduce bugs in the code.
+-   **Scalability** - Contributed FOSS is more scalable than one-off solutions and can grow with the power of the FOSS community.
+-   **Visibility** - When we release FOSS, CivicAction, our developers and our clients get positive representation as technology leaders and contributors.
+
+## Examples of modules CivicActions built as Contrib First
```

Comment:
Can we add some non Drupal ones also change to 'Example of open source repos ...) Examples:

* https://github.com/GetDKAN/cmsds-open-data-components (maybe link DKAN in general)
* https://github.com/drydockcloud
* This guidebook?
* https://github.com/CivicActions/accessibility
* https://github.com/GSA/openacr

Reply:
CivicActions Accessibility and our guidebook does not seem contrib first but maybe my interpretation is flawed.  

1. It seems quite specific to CivicActions rather than something anyone else can use unaltered.
2. It seems like it was not built for a client out of a client need
3. It seems more like it was past work, amalgamated into new public repos.

Reply:
Ah I think both of these warrant #ca-stories stories ;)

Guidebook used to be a full fork of https://github.com/18F/handbook and on the readme we have this line:

> We'd like to thank the folks at 18F for [their handbook](https://github.com/18F/handbook), from which this documentation was forked. We welcome you to fork our documentation, or use the 18F handbook, which is licensed as CC0.

I also believe folks outside of CA have forked our guidebook.

For accessibility, that was also a full fork of https://github.com/18F/uswds-jekyll where we want to create not just the accessibility site but show how folks can quickly spin up sites on GitHub. Maybe this is a better example: https://github.com/CivicActions/ca-jekyll-vanilla

So yes it is a stretch but they were build from opensource projects to show how someone else can. Folks are welcome then to use our examples to build their own. Meets 1., not 2., and yes it is 3.

Reply:
I added 3 out of the 5.